runtime/resolvers/metrics.go

```diff
@@ -87,16 +88,34 @@ func (r *metricsResolver) Close() error {
 	return nil
 }
 
-func (r *metricsResolver) Cacheable() bool {
-	return r.executor.Cacheable(r.query)
-}
+func (r *metricsResolver) CacheKey(ctx context.Context) ([]byte, bool, error) {
+	// get the underlying executor's cache key
+	key, ok, err := cacheKeyForMetricsView(ctx, r.runtime, r.instanceID, r.query.MetricsView, r.args.Priority)
+	if err != nil {
+		return nil, false, err
+	}
+	if !ok {
+		return nil, false, nil
+	}
+
+	hasher := md5.New()
```

Comment:
This is already being hashed in `runtime/resolver.go` so could we return the full pointer here?

Reply:
The values were already being hashed in the older implementation : https://github.com/rilldata/rill/blob/9061a24267f0b562c5304f0d897813fed88055c7/runtime/resolvers/metrics.go#L94

Since the key is now composed of two things (`cache_key` and `query`) I thought of just using a md5 hasher. 
Do you mean we create a new struct and just return any from `CacheKey` instead of bytes ?

Reply:
I just meant if one way or another it could use the same hasher – e.g. it might return a combined `[]byte` or a `*drivers.Result`, or accept a `io.Writer`, or something like that.

It's not an important comment, certainly you shouldn't do anything weird to achieve this (like returning a struct would probably be weird). Just wondered if there was a clean way to have just one hasher.

Reply:
Okay. Added another way to compute key without hasher.